### PR TITLE
tECDSA signing: Ephemeral key exchange and symmetric keys generation

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/keep-core/issues/3042
Depends on: https://github.com/keep-network/keep-core/pull/3219

Here we introduce the two first phases of the tECDSA signing protocol which are about exchanging ephemeral keys and generating symmetric keys. Symmetric keys are meant to secure P2P communication in later phases of the protocol. Basically, the code was ported from the sibling `pkg/tecdsa/dkg` package that also uses those two phases at the beginning.

### Next steps

This PR is just a part of the entire work regarding the tECDSA signing test loop. Please refer to the description of https://github.com/keep-network/keep-core/issues/3042 for the full roadmap and the current status of the work. At this point, the next step will be introducing the first round of TSS signing.

### CODE REVIEW

File: pkg/tecdsa/signing/marshaling.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	ephemeralPublicKeys, err := marshalPublicKeyMap(epkm.ephemeralPublicKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	return (&pb.EphemeralPublicKeyMessage{
+		SenderID:            uint32(epkm.senderID),
+		EphemeralPublicKeys: ephemeralPublicKeys,
+		SessionID:           epkm.sessionID,
+	}).Marshal()
```
Comment (pdyraga) [inline pkg/tecdsa/signing/marshaling.go:22]: I've just realized the ephemeral public key message in the random beacon has a `receiverID` field that is unused. Can we take the opportunity and clean it up?

File: pkg/tecdsa/signing/protocol_test.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+		if ok {
+			t.Errorf(
+				"[member:%v] found ephemeral key pair generated to self",
+				member.id,
+			)
+		}
+
+		// Assert key pairs are non-nil.
+		for otherMemberID, keyPair := range member.ephemeralKeyPairs {
+			if keyPair.PrivateKey == nil {
```
Comment (pdyraga) [inline pkg/tecdsa/signing/protocol_test.go:67]: Should we also assert that `keyPair` is not nil?

File: pkg/tecdsa/signing/protocol_test.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+	// Assert that each message is formed correctly.
+	for memberID, message := range messages {
+		// We should always be the sender of our own messages.
+		testutils.AssertIntsEqual(
+			t,
+			"message sender",
+			int(memberID),
+			int(message.senderID),
+		)
+
```
Comment (pdyraga) [inline pkg/tecdsa/signing/protocol_test.go:94]: Should we assert `groupSize - 1` public keys are generated in each message?

**Review by pdyraga — APPROVED** [review]

File: pkg/tecdsa/signing/marshaling.go (hunk 1 above)
Comment (lukasz-zimnoch) [reply pkg/tecdsa/signing/marshaling.go:22]: Addressed in https://github.com/keep-network/keep-core/pull/3228/commits/a4e17c9bec28d2bc117652e7a869132d384dc48c

File: pkg/tecdsa/signing/protocol_test.go (hunk 3 above)
Comment (lukasz-zimnoch) [reply pkg/tecdsa/signing/protocol_test.go:94]: Addressed in https://github.com/keep-network/keep-core/pull/3228/commits/c0fca3b97e2908eb5d6956a825aa141836d8951a

**Pull request merged**